registry/service.go

```diff
@@ -26,6 +26,7 @@ type Service interface {
 	LoadAllowNondistributableArtifacts([]string) error
 	LoadMirrors([]string) error
 	LoadInsecureRegistries([]string) error
+	IsInsecureRegistry(string) bool
 }
```

Comment:
Yay, another interface with a single implementation. Can we kill it?

Reply:
This exposes a private function of a private (alias) type from the `registry` package, removing this method from the interface would add a lot of noise and feels like out of scope for this PR

Reply:
[I mean delete `type Service interface`](https://github.com/golang/go/wiki/CodeReviewComments#interfaces). But never mind; I see now that there's at least one test-fake implementation which `gopls` "helpfully" neglected to tell me about.

Reply:
Yeah even deleting the interface is out of scope for this PR for me, I could do it in a followup though